### PR TITLE
KFSPTS-12523 upgrade to the 7/19/2018 version of financials

### PR DESCRIPTION
**Pull request by db483**

### CODE REVIEW

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
retest this please

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java [hunk 1]
```diff
@@ -34,7 +36,7 @@
 
 
 public class ProcurementCardCreateDocumentServiceImpl extends org.kuali.kfs.fp.batch.service.impl.ProcurementCardCreateDocumentServiceImpl {
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java]: This class's "createProcurementCardDocument" method had a small amount of formatting and loop iteration changes for the 07/19 patch. Is it worth updating our implementation of the method at this time?

**Comment by db483** [comment]
Chad, 

Thank you for reviewing this:
[1] I missed that, I will add in the changes.
[2] Will add in the changes
[3] The FINP jira states:
2.1. Updating partner project code is optional due to log4j-1.2-api which is a bridge from version 1 to version 2. However, the property file syntax will need to be updated despite the bridge. We have not tested this way of running the code. 

I need to further clarify what/if any changes are needed on our side.
Thank you for pointing this out.

File: src/main/java/edu/cornell/kfs/sys/service/impl/CuBatchFileDirectoryServiceImpl.java [hunk 2]
```diff
@@ -1,20 +1,21 @@
...
 import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
-public class BatchFileDirectoryServiceImpl implements BatchFileDirectoryService {
+public class CuBatchFileDirectoryServiceImpl extends BatchFileDirectoryServiceImpl implements CuBatchFileDirectoryService {
 
-    private static final Logger LOG = Logger.getLogger(BatchFileDirectoryServiceImpl.class);
+    private static final Logger LOG = Logger.getLogger(CuBatchFileDirectoryServiceImpl.class);
```
Comment (cah292) [inline src/main/java/edu/cornell/kfs/sys/service/impl/CuBatchFileDirectoryServiceImpl.java]: This logger still needs to be switched over to the newer one.

File: src/main/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java (hunk 1 above)
Comment (db483) [reply src/main/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java]: updated

File: src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherNonResidentAlienInformationValidation.java [hunk 3]
```diff
@@ -21,14 +23,14 @@
 import org.kuali.kfs.krad.util.MessageMap;
 
 public class CuDisbursementVoucherNonResidentAlienInformationValidation extends DisbursementVoucherNonResidentAlienInformationValidation{
```
[the remark this answers is not in the document]
Comment (db483) [reply src/main/java/edu/cornell/kfs/fp/document/validation/impl/CuDisbursementVoucherNonResidentAlienInformationValidation.java]: added

File: src/test/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImplTest.java [hunk 4]
```diff
@@ -23,7 +25,7 @@
     private ProcurementCardCreateDocumentService procurementCardCreateDocumentService;
     private ConfigurationService  kualiConfigurationService;
     
-    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(ProcurementCardSummaryFeedService.class);
+    private static final Logger LOG = LogManager.getLogger(ProcurementCardSummaryFeedService.class);
```
Comment (cah292) [inline src/test/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImplTest.java]: Should this logger be referencing the unit test class instead? (There appear to be several other unit test classes that also have this mismatched-class setup for their logging.)

Comment (db483) [reply src/test/java/edu/cornell/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImplTest.java]: hmm, good question, I think it should probably be the unit test

**Comment by db483** [comment]
Hi Chad,

I've made the changes you suggested. This is ready for review again. Thank you!

**Comment by db483** [comment]
retest this please

**Comment by db483** [comment]
Thank you Chad, I have talked with Jay and Andrew and  #627 will wait for this to be merged.

**Comment by cah292** [comment]
Just to confirm, are you expecting to have to resolve merge conflicts when this patch gets merged to the master branch? If this PR gets merged prior to the #627 revert, then there will be merge conflicts on the master cherry-pick. Not sure about which other specific changes could have conflicts.

**Comment by db483** [comment]
Chad,

Thank you for pointing this out. It depends on the cherry pick order. There were also some payment works changes affected by this PR so there could be cherry pick conflicts with that.

In the case of  #627 it would be cleaner to merge the revert first and then rebase this. I have mentioned it to Jay.

**Comment by jhulslander** [comment]
retest this please

**Comment by db483** [comment]
I have rebased the PR and this is ready for review again.

Thank you.

**Pull request merged**